asreview/balance_strategies/base.py

```diff
@@ -12,35 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from abc import ABC, abstractmethod
+from abc import abstractmethod
+from asreview.base_model import BaseModel
 
-from asreview.utils import _unsafe_dict_update
 
+class BaseBalance(BaseModel):
+    "Abstract class for balance strategies."
+    name = "base-balance"
 
-class BaseTrainData(ABC):
-    " Abstract class for balance strategies. "
-    def __init__(self, balance_kwargs):
-        self.balance_kwargs = self.default_kwargs()
-        self.balance_kwargs = _unsafe_dict_update(self.balance_kwargs,
-                                                  balance_kwargs)
-
-    def func_kwargs_descr(self):
-        " Should give back the function and arguments for balancing. "
-        return (self.__class__.function(), self.balance_kwargs,
-                self.__class__.description())
-
-    def default_kwargs(self):
-        return {}
-
-    def hyperopt_space(self):
-        return {}
-
-    @staticmethod
-    @abstractmethod
-    def function():
-        raise NotImplementedError
-
-    @staticmethod
     @abstractmethod
-    def description():
+    def sample(self, X, y, train_idx, shared):
```

Comment:
shared -> make use of kwargs?

Reply:
I'm not sure, but I *think* I prefer it like this, since it ensures that the shared dictionary is passed on (in case the balance model needs it).